File: src/datastructures/bst/BST.java
package datastructures.bst;

public class BST {
    Node root;

    class Node {
        int value;
        Node right;
        Node left;

        private Node(int value) {
            this.value = value;
        }
    }

    public boolean insert(int value) {
        // create newNode
        Node newNode = new Node(value);
        // if root == null then root = newNode
        if (root == null) {
            root = newNode;
            return true;
        }
        // temp = root
        Node temp = root;
        // while loop
        while (true) {
            // if newNode == temp return false
            if (newNode.value == temp.value) {
                return false;
            }
            // if < left else > right
            if (newNode.value < temp.value) {
                // if null insert newNode else move to next
                if (temp.left == null) {
                    temp.left = newNode;
                    return true;
                } else {
                    temp = temp.left;
                }
            } else if (newNode.value > temp.value) {
                // if null insert newNode else move to next
                if (temp.right == null) {
                    temp.right = newNode;
                    return true;
                } else {
                    temp = temp.right;
                }
            }
        }
    }

    public boolean contains(int value) {
        Node temp = root;
        while (temp != null) {
            if (value == temp.value) {
                return true;
            }
            if (value < temp.value) {
                temp = temp.left;
            } else if (value > temp.value) {
                temp = temp.right;
            } else {
                return true;
            }
        }
        return false;
    }
}

File: src/datastructures/bst/BST.py
class Node:
    def __init__(self, value):
        self.value: int = value
        self.left: Node = None
        self.right: Node = None


class BST:
    def __init__(self):
        self.root: Node = None

    def insert(self, value: int) -> bool:
        newNode: Node = Node(value)
        if self.root == None:
            self.root = newNode
            return True
        temp: Node = self.root
        while True:
            if newNode.value == temp.value:
                return False
            if newNode.value < temp.value:
                if temp.left == None:
                    temp.left = newNode
                    return True
                else:
                    temp = temp.left
            if newNode.value > temp.value:
                if temp.right == None:
                    temp.right = newNode
                    return True
                else:
                    temp = temp.right

    def contains(self, value: int) -> bool:
        if self.root == None:
            return False
        temp: Node = self.root
        while temp != None:
            if value == temp.value:
                return True
            if value < temp.value:
                temp = temp.left
            elif value > temp.value:
                temp = temp.right
            else:
                return True
        return False


myBST = BST()
myBST.insert(47)
myBST.insert(21)
myBST.insert(76)
myBST.insert(18)
myBST.insert(52)
myBST.insert(82)

myBST.insert(27)

print(myBST.root.left.right.value)
print(myBST.contains(27))
print(myBST.contains(26))
